### PR TITLE
fix: bug preventing transactions from working

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

* Fixes static fee transactions.
* Fixes dynamic fee transactions.

### How I did it

Build our own RLP serializable classes.

### How to verify it

```python
from ape import accounts, project
from ape.logging import logger


def main():
    logger.info("Starting simulation...")

    funder = accounts.test_accounts[5]

    try:
        account = accounts.load("main")
    except IndexError:
        logger.error("Ledger plugin has likely failed to load :(")
        return

    logger.info("Funding main account...")
    funder.transfer(account, "900 ETH")
    logger.success("Funding complete!")
    assert account.balance

    logger.info("Deploying smart contract using Static txn...")
    account.deploy(project.Fund, type=0)
    logger.success("Smart contract deployed!")

    logger.info("Deploying smart contract using Dynamic txn...")
    account.deploy(project.Fund, gas_limit=30029122)
    logger.success("Smart contract deployed!")

    logger.info("Returning funds to funder statically...")
    account.transfer(funder, "400 ETH", type=0)
    logger.success("Refunding complete!")
    assert account.balance

    logger.info("Returning funds to funder statically...")
    account.transfer(funder, "400 ETH")
    logger.success("Refunding complete!")
    assert account.balance

    logger.success("Simulation complete!")
```

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
tried it with fantom and got:
```py
eth_account/_utils/signing.py:95, in extract_chain_id(raw_v)
     94     else:
---> 95         raise ValueError("v %r is invalid, must be one of: 0, 1, 27, 28, 35+")
     96 else:

ValueError: v %r is invalid, must be one of: 0, 1, 27, 28, 35+
```

**Comment by unparalleled-js** [comment]
I really want to stop using `eth-account` altogether for this

**Comment by unparalleled-js** [comment]
I am close to having something that works great for both dynamic and static! Hopefully by EOD

**Review by fubuloubu — APPROVED** [review]

File: ape_ledger/objects.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+from typing import Optional, Union
+
+from eth_typing import HexStr
+from eth_utils import add_0x_prefix, decode_hex
+from rlp import Serializable  # type: ignore
+from rlp.sedes import Binary  # type: ignore
+from rlp.sedes import List as ListSedes  # type: ignore
+from rlp.sedes import big_endian_int, binary  # type: ignore
+
+"""Inspired from https://github.com/unparalleled-js/ledger-eth-lib"""
```
[the remark this answers is not in the document]
Comment (unparalleled-js) [reply ape_ledger/objects.py:10]: Oops, yes thank you

**Comment by unparalleled-js** [comment]
Things still ain't quite right but investigating

File: ape_ledger/accounts.py [hunk 2]
```diff
@@ -84,12 +88,55 @@ def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
...
         else:
             raise LedgerSigningError(
-                f"Unsupported message-signing specification, (version={version!r})"
+                f"Unsupported message-signing specification, (version={version!r})."
             )
 
-        return MessageSignature(*signed_msg)  # type: ignore
+        v, r, s = signed_msg
+
+        if self.provider:
```
Comment (unparalleled-js) [inline ape_ledger/accounts.py:96]: I am not happy with this part but I don't know how else to do it.
Basically, it seems like we need to do some modulo math on `v` for some chain IDs.

https://github.com/LedgerHQ/app-ethereum/blob/develop/examples/signMessageEIP711v0.py#L75-L81
https://github.com/LedgerHQ/app-ethereum/blob/develop/examples/signMessage.py#L75-L81

So we need to know the chain ID at this point, but the user is not connected to a provider, who knows what they heck they are signing....

At least in `sign_transaction`, chain ID is set on the transaction. Here, we can't assume the chain ID is in the message anywhere so we have to check the active provider.

Comment (fubuloubu) [reply ape_ledger/accounts.py:96]: This logic seems pretty appropiate to me actually

**Comment by unparalleled-js** [comment]
@fubuloubu Are you good with merging this? I want to get these changes in the pydantic PR.

We can test Fantom support more when `ape-fantom` is ready. If `v` still bugs out, we can create a new issue then

**Comment by fubuloubu** [comment]
> @fubuloubu Are you good with merging this? I want to get these changes in the pydantic PR.
> 
> We can test Fantom support more when `ape-fantom` is ready. If `v` still bugs out, we can create a new issue then

Did you want this to work with latest `ape` main?

```py
$ ape ledger list
WARNING: 'AccountContainer - LedgerAccount' from 'ape_ledger' is not fully implemented. Remaining abstract methods: 'accounts'.
```

**Review by fubuloubu — APPROVED** [review]
Works locally (trying with `ape-fantom` next)

**Pull request merged**